src/gt4py/next/ffront/decorator.py

```diff
@@ -259,6 +264,9 @@ def __call__(self, *args: Any, offset_provider: common.OffsetProvider, **kwargs:
                 ctx.run(self.definition_stage.definition, *args, **kwargs)
             return
 
+        if config.COLLECT_METRICS:
+            offset_provider["exec_info"] = metrics.global_metric_container[self.__name__]
```

Comment:
the only reason why I didn't get rid of this hack is that I don't have the name available in the gtfn run...